lib/node_loader.js

```diff
@@ -90,10 +88,9 @@ AWS.util.update(AWS.Config.prototype.keys, {
         {isConfig: true, filename: env[AWS.util.sharedConfigFileEnv]}
       ];
       while (!region && toCheck.length) {
-        var configFile = new SharedIniFile(toCheck.shift());
-        var profile = configFile.getProfile(
-          env.AWS_PROFILE || AWS.util.defaultProfile
-        );
+        var iniLoader = require('./shared-ini');
```

Comment:
I know iniLoader is cached, but I think it still makes sense to put it outside of the while loop since you're using a single instance.